### ISSUE TITLE
[Request] Some features to improve Flairmax

### ISSUE THREAD

**Issue by realgooseman** [flairmax/FlairMax-Issues#6]
Hi @ahmed605,

After searching for a Windows alternative to Mousai (Linux only) I finally found Flairmax. Works great is lightweight and looks good.

But here are my suggestions to improve the app and get more people to discover it :

**1.** Flairmax was quite hard to find, so I'd suggest submitting the app on AlternativeTo or equivalent for faster discovery by people.
**2.** A dark theme for the app would be great, maybe with Windows system accent color too.
**2.** The theme looks like Windows 10, maybe refreshing the interface to WinUI 3 to better fit modern Windows 11 style ?
**3.** App translation in multiple languages (via Crowdin ?), I can help with French and Dutch.

All these are improvements that came to my mind when using Flairmax for a bit, let me know what you think.

Unrelated but do you ever plan to open source the app and have a proper GitHub repo with releases section ?

**Comment by ahmed605** [comment]
Hi, @realgooseman!

**1.** Good point, will give AlternativeTo a try, thanks for the suggestion.
**2.** That's planned, yeah.
**3.** What parts of the UI look like Windows 10 to you? the app uses its own design so it's neither Win10 or Win11, I mean the design is inspired from original Fluent Design but it's still modified to be different and unique but let me know which parts needs improvement. (just to clarify, I'm against using Fluent Design 2.0 (Win11 design) as-is because I want t have a unique design but I'll happily accept feedbacks suggesting improvements to that design).
**4.** That's also planned but will take a while to implement but thanks for your offer, will give you a ping when it's ready. 


> Unrelated but do you ever plan to open source the app and have a proper GitHub repo with releases section?

Open sourcing isn't planned but I'm actually working on a repo (currently private) called `FlairMax.OpenSource` that would include parts from the source that I think will be useful to other devs.

Thank you very much for your feedback!

**Comment by realgooseman** [comment]
Great to know that a dark mode and translations are planned. 

**Some UI/UX improvements :**

- I personally find the outline buttons unappealing. Filled buttons with accent color would look better.

- Home button on scanned song page to go back to start page

- Option to remove blurred (Mica/Acrylic) background for scanned song page and only keep light/dark theme background

- Option to hide some "Listen to this song on" options (Hide Deezer, Apple Music for example) + make the links appear faster because they're quite slow to load for now.

- Changing the placement of top right buttons to keep it simple. Only leave lyrics, download track and history buttons. The rest goes in the three dotted dropdown menu with settings.

- Instead of having a dropdown menu to choose from Title, Artist, Album + copy to clipboard button. Just make the "Copy to clipboard" button a dropdown menu with those same options for faster copying. Would remove the need of two separate elements for the same thing.

- Make scanned songs in history page a little bit larger so less song titles get cut-off (changing from square shape to a little bit more rectangular)

**Comment by dongle-the-gadget** [comment]
> Option to hide some "Listen to this song on" options (Hide Deezer, Apple Music for example) + make the links appear faster because they're quite slow to load for now.

As far as I’m aware, that delay is mostly caused by the fact that the service which FlairMax uses is slow rather than UI performance

**Comment by realgooseman** [comment]
Yes, I think so too. And doesn't always find links for scanned songs. But it works most of the time so it's good for now.

Forgot this issue, any updates on FlairMax's development @ahmed605 ?
I got a new update a few days ago, but not much changed.

**Comment by ahmed605** [comment]
> Yes, I think so too. And doesn't always find links for scanned songs. But it works most of the time so it's good for now.
> 
> Forgot this issue, any updates on FlairMax's development @ahmed605 ?
> I got a new update a few days ago, but not much changed.

I didn't have the time to share the changelog everywhere yet (will do once I get the time) but here it is:

- **Audio Visualizers**: 4 new audio visualizers to use as a background for Synced Lyrics, in addition to the Cover Art and the Animated Cover Art ones. (powered by [ComputeSharp](https://github.com/Sergio0694/ComputeSharp)).
- **Exporting History**: you can now export your recognition history as a file, the current format is `[Artist] - [Title]` but it's planned to be improved in future releases.
- **Run while minimized**: provides the ability to continue recognizing audio tracks while the app is minimized, this feature is off by default but can be turned on in Settings.
- **Adaptive Layout:** the layout is now much more adaptive especially on touch screens and small window sizes.
- **Improved YouTube Playback**: YouTube playback should be much more reliable and faster in this release, this release also includes support for 2k & 4k videos and higher quality audio for YouTube videos.
- **Improved Shazam Core**: Shazam Core got many improvements in this release and should be faster and more accurate now.
- **Improved "Scrolling" Synced Lyrics Effect:** the Scrolling effect on build 17763 (and older) should be on par in quality with how it was on builds newer than 18362 in the previous release, so it should look & feel much better on these builds.
- **Startup Optimizations**: app startup should be faster now, and instant if the app was preloaded at least once before.
- **UI Improvements**: various UI tweaks including using Segoe UI Variable everywhere when running Windows 11 (yes, that was disabled before).
- **Other small improvements**: various bug & crash fixes, and better experience on ARM64 devices.

**Comment by realgooseman** [comment]
Great additions! and it's good to see that FlairMax is still being actively developed and maintained. There simply isn't a good enough alternative, the main one (MRA) got caught with a shady installer that includes malware so pretty much FlairMax only on Windows systems.

Hoping a dark mode and localization is still considered for the future.

**Comment by ahmed605** [comment]
yep, they are still planned, hopefully they will ship in the next update if everything went as expected

**Comment by realgooseman** [comment]
Hi @ahmed605, 

It's been over a year since I posted this issue. Any updates on Flairmax ?
It's not open sourced so can't see the commits and work. Is it still maintained or abandoned ?

**Comment by ahmed605** [comment]
> Hi @ahmed605, 
> 
> It's been over a year since I posted this issue. Any updates on Flairmax ?
> It's not open sourced so can't see the commits and work. Is it still maintained or abandoned ?

Hi
It's still maintained, yes, I did few commits this month and there are multiple finished features done but still unreleased, it's just going slow because I don't have much time nowadays but it's still alive

**Comment by realgooseman** [comment]
Great to hear, understandable no rush.
I look forward to future releases and improvements.
